### PR TITLE
MariaDB configuration documentation in English and French

### PR DESCRIPTION
**Pull request by dooremont**
##### Checklist

-   [ ] the [individual contributor license agreement][] is signed
-   [ ] commit message follows [commit guidelines][]
-   [ ] tests are included
-   [x] documentation is changed or added
-   [ ] [message properties][] have been updated with new phrases
-   [ ] view conforms with [WCAG 2.0 AA][]

##### Description of change

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: docs/database/mariadb.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+This database configuration is not suitable for production deployments but is better suited for testing purposes.**
+
+uPortal supports a number of production databases and you can configure the MariaDB database.
+
+## Step 1: MariaDB server setup
+Edit the file /etc/mysql/mariadb.conf.d/50-server.cnf. (Debian 9)
+In the [mysqld] part add the following items :
+
+```shell
```
Comment (ChristianMurphy) [inline docs/database/mariadb.md]: using the flag
`` ```properties ``
will ensure accurate syntax highlighting.

File: docs/database/mariadb.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+MariaDB [(none)]> create database uportal CHARACTER SET utf8 COLLATE utf8_general_ci;
+MariaDB [(none)]> create database portlets CHARACTER SET utf8 COLLATE utf8_general_ci;
+CREATE USER 'uportal'@'localhost' IDENTIFIED BY 'uportal';
+GRANT ALL PRIVILEGES ON uportal.* TO 'portail'@'localhost';
+GRANT ALL PRIVILEGES ON portlets.* TO 'portail'@'localhost';
+```
+## Step 3: Configure Uportal 
+
+# Edit uPortal-start/gradle.properties 
+```shell
```
Comment (ChristianMurphy) [inline docs/database/mariadb.md]: flag `properties`

File: docs/database/mariadb.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+mysqldbVersion=5.1.45
+```
+# Edit uPortal-start/overlays/build.gradle
+```shell
```
Comment (ChristianMurphy) [inline docs/database/mariadb.md]: flag `gradle`

File: docs/database/mariadb.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+         * These are nearly the same uPortal dependencies declared by uPortal-webapp;
+         * perhaps we should create a uPortal-all module to bundle them all as transitives.
+         */
+
+```
+
+# Edit uPortal-start/etc/portal/global.properties 
+
+In the Database Connection section
+```shell
```
Comment (ChristianMurphy) [inline docs/database/mariadb.md]: flag `properties`

File: docs/database/mariadb.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+hibernate.connection.driver_class=com.mysql.jdbc.Driver
+hibernate.connection.url=jdbc:mysql://localhost/portlets
+hibernate.connection.username=uportal
+hibernate.connection.password=uportal
+hibernate.connection.validationQuery=select 1
+hibernate.dialect = org.hibernate.dialect.MySQL5InnoDBDialect
+```
+# Edit uPortal-start/etc/portal/uPortal.properties
+
+```shell
```
Comment (ChristianMurphy) [inline docs/database/mariadb.md]: flag `properties`

File: docs/fr/database/mariadb.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+**Cette configuration de base de données ne convient pas aux déploiements de production mais est mieux adaptée à des fins de test.**
+
+uPortal prend en charge un certain nombre de bases de données de production et vous pouvez configurer la base de données MariaDB.
+
+## Étape 1 : Paramétrage du server MariaDB
+Editer le fichier /etc/mysql/mariadb.conf.d/50-server.cnf. (ici pour Debian 9)
+Dans la partie mysqld ajouter les éléements suivant :
+
+```shell
```
Comment (ChristianMurphy) [inline docs/fr/database/mariadb.md]: flag `properties`

File: docs/fr/database/mariadb.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+MariaDB [(none)]> create database uportal CHARACTER SET utf8 COLLATE utf8_general_ci;
+MariaDB [(none)]> create database portlets CHARACTER SET utf8 COLLATE utf8_general_ci;
+CREATE USER 'uportal'@'localhost' IDENTIFIED BY 'uportal';
+GRANT ALL PRIVILEGES ON uportal.* TO 'portail'@'localhost';
+GRANT ALL PRIVILEGES ON portlets.* TO 'portail'@'localhost';
+```
+## Étape 3 : Configurer Uportal 
+
+# Editer uPortal-start/gradle.properties 
+```shell
```
Comment (ChristianMurphy) [inline docs/fr/database/mariadb.md]: flag `properties`

File: docs/fr/database/mariadb.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+mysqldbVersion=5.1.45
+```
+# Editer uPortal-start/overlays/build.gradle
+```shell
```
Comment (ChristianMurphy) [inline docs/fr/database/mariadb.md]: flag `gradle`

File: docs/fr/database/mariadb.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+         * These are nearly the same uPortal dependencies declared by uPortal-webapp;
+         * perhaps we should create a uPortal-all module to bundle them all as transitives.
+         */
+
+```
+
+# Editer uPortal-start/etc/portal/global.properties 
+
+Dans la partie Database Connection
+```shell
```
Comment (ChristianMurphy) [inline docs/fr/database/mariadb.md]: flag `properties`

File: docs/fr/database/mariadb.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+hibernate.connection.driver_class=com.mysql.jdbc.Driver
+hibernate.connection.url=jdbc:mysql://localhost/portlets
+hibernate.connection.username=uportal
+hibernate.connection.password=uportal
+hibernate.connection.validationQuery=select 1
+hibernate.dialect = org.hibernate.dialect.MySQL5InnoDBDialect
+```
+# Editer uPortal-start/etc/portal/uPortal.properties
+
+```shell
```
Comment (ChristianMurphy) [inline docs/fr/database/mariadb.md]: flag `properties`

File: docs/database/mariadb.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+hibernate.connection.url=jdbc:mysql://localhost/uportal
+hibernate.connection.username=uportal
+hibernate.connection.password=uportal
+hibernate.connection.validationQuery=select 1
+hibernate.dialect = org.hibernate.dialect.MySQL5InnoDBDialect
+```
+
+## Étape 4 : Initialization of the Database
+
+./gradlew dataInit
```
Comment (ChristianMurphy) [inline docs/database/mariadb.md]: wrap in code fence

    ```shell
    ./gradlew dataInit
    ```

File: docs/database/mariadb.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+## Étape 5 : Deployment of uPortal
+
+./gradlew tomcatDeploy
```
Comment (ChristianMurphy) [inline docs/database/mariadb.md]: wrap in code fence

    ```shell
    ./gradlew tomcatDeploy
    ```

File: docs/fr/database/mariadb.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+hibernate.dialect = org.hibernate.dialect.MySQL5InnoDBDialect
+```
+
+## Étape 4 : Initialisation de la Base de Donnée
+
+./gradlew dataInit
+
+## Étape 5 : Déploiement de uPortal
+
+./gradlew tomcatDeploy
```
Comment (ChristianMurphy) [inline docs/fr/database/mariadb.md]: wrap in code fence

    ```shell
    ./gradlew tomcatDeploy
    ```

File: docs/fr/database/mariadb.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+hibernate.connection.url=jdbc:mysql://localhost/uportal
+hibernate.connection.username=uportal
+hibernate.connection.password=uportal
+hibernate.connection.validationQuery=select 1
+hibernate.dialect = org.hibernate.dialect.MySQL5InnoDBDialect
+```
+
+## Étape 4 : Initialisation de la Base de Donnée
+
+./gradlew dataInit
```
Comment (ChristianMurphy) [inline docs/fr/database/mariadb.md]: wrap in code fence

    ```shell
    ./gradlew dataInit
    ```

File: docs/fr/database/mariadb.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+
+uPortal est configuré pour utiliser une base de données HSQL par défaut.
+
+**Cette configuration de base de données ne convient pas aux déploiements de production mais est mieux adaptée à des fins de test.**
+
+uPortal prend en charge un certain nombre de bases de données de production et vous pouvez configurer la base de données MariaDB.
+
+## Étape 1 : Paramétrage du server MariaDB
+Editer le fichier /etc/mysql/mariadb.conf.d/50-server.cnf. (ici pour Debian 9)
+Dans la partie mysqld ajouter les éléements suivant :
```
Comment (jgribonvald) [inline docs/fr/database/mariadb.md]: just one typo : éléments in place of éléements

File: docs/fr/database/mariadb.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
+# Configuration de la Base de Données MariaDB
+
+uPortal est configuré pour utiliser une base de données HSQL par défaut.
+
+**Cette configuration de base de données ne convient pas aux déploiements de production mais est mieux adaptée à des fins de test.**
+
+uPortal prend en charge un certain nombre de bases de données de production et vous pouvez configurer la base de données MariaDB.
+
+## Étape 1 : Paramétrage du server MariaDB
+Editer le fichier /etc/mysql/mariadb.conf.d/50-server.cnf. (ici pour Debian 9)
```
Comment (jgribonvald) [inline docs/fr/database/mariadb.md]: I would suggested to copy 50-server.cnf into 60-server.cnf as conf are read in filename order, remove useless properties (as read from default files) and adapt only yours ;) but for the doc it's ok

**Comment by jgribonvald** [comment]
Thanks !

**Comment by jgribonvald** [comment]
I have to purpose a PR to solve the little problem on dbinit when using mariaDB connector (I know where)
=> I will be able to do it next week ;)

**Comment by bjagg** [comment]
:+1: and welcome aboard! So happy to have new contributors submit PRs!! :1st_place_medal:

**Pull request merged**